src/github.com/matrix-org/dendrite/appservice/consumers/roomserver.go

```diff
@@ -113,6 +116,10 @@ func (s *OutputRoomEventConsumer) onMessage(msg *sarama.ConsumerMessage) error {
 	}
 
 	// Check if any events need to passed on to external application services
+	if len(events) > 0 {
+		// Check if this was a membership event
+		return s.filterRoomserverEvents(ctx, events)
```

Comment:
What is this doing? Why are we returning? The fact that we then call this again with `append(events, ev)` after feels weird.

Reply:
I'm also not sure we want to be resending state events to the AS. Does synapse do this?

Reply:
It apparently does. And yeah, I agree the formatting here is a bit weird, will fix up.